tests/test_flake8_trio.py

```diff
@@ -465,12 +467,42 @@ async def foo():
     sync_fns.takes_a_long_time()
 """
     )
+    return (
+        "./example.py:5:5: TRIO200: User-configured blocking sync call sync_fns.* "
+        "in async function, consider replacing with the_async_equivalent.\n"
+    )
+
+
+def test_200_from_config_flake8_internals(
+    tmp_path: Path, capsys: pytest.CaptureFixture[str]
+):
+    # abuse flake8 internals to avoid having to use subprocess
+    # which breaks breakpoints and hinders debugging
+    # TODO: fixture (?) to change working directory
+
+    err_msg = _test_trio200_from_config_common(tmp_path)
+    # replace ./ with tmp_path/
+    err_msg = str(tmp_path) + err_msg[1:]
+
+    from flake8.main.cli import main
+
+    main(
+        argv=[
+            str(tmp_path / "example.py"),
+            "--append-config",
+            str(tmp_path / ".flake8"),
+        ]
+    )
+    out, err = capsys.readouterr()
+    assert not err
+    assert err_msg == out
```

Comment:
almost funny after the fact how massively the `subprocess` call broke my debugging workflow and made finding out why tests broke take ages (a trailing comma in the list of config options).

I broke this out as a separate test in case flake8 internals break, but feels like there should be a cleaner solution that doesn't rely on any internal structure at all and just emulates calling it with `python -m`. Something like `runpy.run_module` but with also setting `sys.argv` and working dir. I didn't find anything that didn't require writing a bunch of extra code, but if you have anything I'd love to hear of it.

Reply:
No specific advice sorry, but I appreciate having this test broken out!